DOCS-7855-7856-4.3-Changes-for-SDK (#63)
mulesoft/docs-mule-sdk: modules/ROOT/pages/metadata-keys.adoc

  public void query(@MetadataKeyId String query){
      //...
  }
----

> == Optional MetadataKey

Reviewer: ```suggestion
== Optional MetadataKey Parameters
```

Available since version 1.3.

> MetadataKey parameters can be made optional like the other parameters by adding the ``@Optional` annotation. For example:

Reviewer: ditto re. passive

Reviewer: ```suggestion
You can make `MetadataKey` parameters optional in the same way as you do other parameters, by adding the `@Optional` annotation:
```

Author: ```suggestion
You can make `MetadataKey` parameters optional like other parameters, by adding the `@Optional` annotation:
```

[source,java,linenums]
----
public void query(@MetadataKeyId @Optional String query){
 //...
}
----

> When an optional MetadataKey parameter does not have a value, metadata calls are made using null as MetadataKey. This means that any MetadataResolver associated with this MetadataKey must be prepared to receive a null value as MetadataKey.

Reviewer: ```suggestion
When an optional MetadataKey parameter does not have a value, metadata calls are made using `null` as the value of `MetadataKey`. This means that any `MetadataResolver` parameter associated with this `MetadataKey` parameter must be prepared to receive a null value.
```

Reviewer: As requested earlier, please search for and properly format syntax elements, per our style ref. entry "font choices."
I'll stop trying to find and mark them all.

Author: ```suggestion
When an optional MetadataKey parameter is not assigned a value, metadata calls are made using `null` as the value of `MetadataKey`. This means that any `MetadataResolver` parameter associated with this `MetadataKey` parameter must be prepared to receive a null value.
```
